sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -0,0 +1,314 @@
+        self._working_threads = {}  # type: Dict[str, threading.Thread]
+        self._threads_stop_flags = {}  # type: Dict[str, bool]
+
+        self._callback_queue = queue.Queue(maxsize=10000)  # Right now the limitation of receiving speed is ~10k
+
+    def __repr__(self):
+        return 'EventProcessor: id {}'.format(self._id)
+
+    def _cancel_tasks_for_partitions(self, to_cancel_partitions):
+        to_stop_partition_threads = []
+        with self._lock:
+            for partition_id in to_cancel_partitions:
+                if partition_id in self._working_threads:
+                    thread = self._working_threads.pop(partition_id)
+                    self._threads_stop_flags[partition_id] = True
+                    to_stop_partition_threads.append((partition_id, thread))
+            for partition_id, thread in to_stop_partition_threads:
+                thread.join()
+                del self._threads_stop_flags[partition_id]
+        if to_cancel_partitions:
+            log.info("EventProcesor %r has cancelled partitions %r", self._id, to_cancel_partitions)
+
+    def _create_tasks_for_claimed_ownership(self, claimed_partitions, checkpoints=None):
+        with self._lock:
+            for partition_id in claimed_partitions:
+                checkpoint = checkpoints.get(partition_id) if checkpoints else None
+                if partition_id not in self._working_threads or not self._working_threads[partition_id].is_alive():
+                    self._working_threads[partition_id] = threading.Thread(target=self._receive,
+                                                                           args=(partition_id, checkpoint))
+                    self._working_threads[partition_id].daemon = True
+                    self._threads_stop_flags[partition_id] = False
+                    self._working_threads[partition_id].start()
+                    log.info("Working thread started, ownership %r, checkpoint %r", partition_id, checkpoint)
+
+    @contextmanager
+    def _context(self, events):
+        # Tracing
+        span_impl_type = settings.tracing_implementation()  # type: Type[AbstractSpan]
+        if span_impl_type is None:
+            yield
+        else:
+            child = span_impl_type(name="Azure.EventHubs.process")
+            self._eventhub_client._add_span_request_attributes(child)  # pylint: disable=protected-access
+            child.kind = SpanKind.SERVER
+
+            for event in events:
+                event._trace_link_message(child)  # pylint: disable=protected-access
+            with child:
+                yield
+
+    def _receive(self, partition_id, checkpoint=None):  # pylint: disable=too-many-statements
+        log.info("start ownership %r, checkpoint %r", partition_id, checkpoint)
+        namespace = self._namespace
+        eventhub_name = self._eventhub_name
+        consumer_group_name = self._consumer_group_name
+        owner_id = self._id
```

Comment:
Why are these 4 assigned to local variables? Seems that we can just use the attributes directly?